### PR TITLE
Skip core pack installation in XSIAM

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-7163)

## Description
- Added the **UNREMOVABLE_PACKS** list.
- Updated **reset_base_pack_version** method name to ** reset_core_pack_version**.
- Use the **search_and_install_packs_and_their_dependencies** method to install the **UNREMOVABLE_PACKS**.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/search_and_uninstall_pack.py [hunk 1]
```diff
@@ -35,8 +36,9 @@ def get_all_installed_packs(client: demisto_client):
             installed_packs_ids_str = ', '.join(installed_packs_ids)
             logging.debug(
                 f'The following packs are currently installed from a previous build run:\n{installed_packs_ids_str}')
-            if 'Base' in installed_packs_ids:
-                installed_packs_ids.remove('Base')
+            for pack in UNREMOVABLE_PACKS:
+                if pack in installed_packs_ids:
+                    installed_packs_ids.remove('Base')
```
Comment (daryakoval) [inline Tests/Marketplace/search_and_uninstall_pack.py:41]:
```suggestion
                    installed_packs_ids.remove(pack)
```

**Review by daryakoval — CHANGES_REQUESTED** [review]
Please handle all "unremoable packs" like the base pack in the function: 

```def reset_base_pack_version(client: demisto_client):```

File: Tests/Marketplace/search_and_uninstall_pack.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 from Tests.Marketplace.search_and_install_packs import install_packs
 from time import sleep
 
+UNREMOVABLE_PACKS = ['Base', 'CoreAlertFields', 'Core']
```
Comment (daryakoval) [inline Tests/Marketplace/search_and_uninstall_pack.py:14]: Are there mandatory dependencies for these packs?

Comment (adi88d) [reply Tests/Marketplace/search_and_uninstall_pack.py:14]: yes, for **Core**:
image
and for **CoreAlertFields**:
image

File: Tests/Marketplace/search_and_uninstall_pack.py [hunk 3]
```diff
@@ -92,48 +95,19 @@ def uninstall_all_packs(client: demisto_client, hostname):
...
-                logging.debug('Found Base pack in bucket!')
-
-                pack_data = {
-                    'id': result_object.get('id'),
-                    'version': result_object.get('currentVersion')
-                }
-                # install latest version of Base pack
-                logging.info(f'updating base pack to version {result_object.get("currentVersion")}')
-                return install_packs(client, host, [pack_data], False)
+    _, success = search_and_install_packs_and_their_dependencies(UNREMOVABLE_PACKS, client, host, True)
```
Comment (daryakoval) [inline Tests/Marketplace/search_and_uninstall_pack.py:108]:
```suggestion
    _, success = search_and_install_packs_and_their_dependencies(pack_ids=UNREMOVABLE_PACKS, client=client, hostname=host, install_packs_one_by_one=True)
```

**Review by daryakoval — APPROVED** [review]

**Pull request merged**